### ISSUE TITLE
[JAX-RS RESTEasy] Generated JAX-RS RESTEasy stub is missing swagger annotations

### ISSUE THREAD

**Issue by karnik** [swagger-api/swagger-codegen#4447]
##### Description

After generating the server stub for my jee test application based on RESTEasy with swagger-codegen the generated swagger.json just shows:

```json
{"swagger":"2.0"}
```

When I checked the generated server stub I noticed that the swagger annotations are missing. This is also the case with the petstore when exporting the stub with the swagger-editor. When generating for the default JAX-RS (Jersey), the code does include the annotations. For Example:

**Pets.java - JAX-RS RESTEasy**

```java
@Path("/pets")
@Consumes({ "application/json" })
@Produces({ "application/json" })
@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaResteasyServerCodegen", date = "2016-12-21T16:38:44.967Z")
public class PetsApi  {
   private final PetsApiService delegate = PetsApiServiceFactory.getPetsApi();

    @POST
    
    @Consumes({ "application/json" })
    @Produces({ "application/json" })
    public Response addPet( NewPet pet,@Context SecurityContext securityContext)
    throws NotFoundException {
        return delegate.addPet(pet,securityContext);
    }
   . . .
}
```

**PetsApi.java - JAX-RS (Jersey)**

```java
@Path("/pets")
@Consumes({ "application/json" })
@Produces({ "application/json" })
@io.swagger.annotations.Api(description = "the pets API")
@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaJerseyServerCodegen", date = "2016-12-21T16:39:30.729Z")
public class PetsApi  {
   private final PetsApiService delegate = PetsApiServiceFactory.getPetsApi();

    @POST
    
    @Consumes({ "application/json" })
    @Produces({ "application/json" })
    @io.swagger.annotations.ApiOperation(value = "", notes = "Creates a new pet in the store.  Duplicates are allowed", response = Pet.class, tags={  })
    @io.swagger.annotations.ApiResponses(value = { 
        @io.swagger.annotations.ApiResponse(code = 200, message = "pet response", response = Pet.class),
        
        @io.swagger.annotations.ApiResponse(code = 200, message = "unexpected error", response = Pet.class) })
    public Response addPet(@ApiParam(value = "Pet to add to the store" ,required=true) NewPet pet
,@Context SecurityContext securityContext)
    throws NotFoundException {
        return delegate.addPet(pet,securityContext);
    }
   . . .
}
```

After checking the templates I found out that there is no swagger annotation definition in the RESTEasy templates at all. 

##### Swagger-codegen version

Maven Plugin: 2.2.1
Swagger core: 1.5.10

##### Swagger declaration file content or url

Used the Swagger Petstore (Simple) definition
[swagger.zip](https://github.com/swagger-api/swagger-codegen/files/666876/swagger.zip)

##### Command line used for generation

Used maven plugin:

```
                <groupId>io.swagger</groupId>
                <artifactId>swagger-codegen-maven-plugin</artifactId>
                <version>2.2.1</version>
```

##### Steps to reproduce

Generate the JAX-RS RESTEasy server stub with the swagger-codegen maven plugin.

##### Related issues

Nothing found.

##### Suggest a Fix
When I add the annotations to the generated code manually they will appear in the generated swagger.json. So adding the annotations to the template should fix this issue!?

**Comment by wing328** [comment]
> When I checked the generated server stub I noticed that the swagger annotations are missing. This is also the case with the petstore when exporting the stub with the swagger-editor. When generating for the default JAX-RS (Jersey), the code does include the annotations.

@karnik I agree with you that JAX-RS (Jersey) is more mature than JAX-RS RESTEasy. May I know if you've time to contribute the enhancement (swagger annotations) for JAX-RS RESTEasy?

Ref: Mustache templates for JAX-RS RESTEasy https://github.com/swagger-api/swagger-codegen/tree/master/modules/swagger-codegen/src/main/resources/JavaJaxRS/resteasy

**Comment by karnik** [comment]
@wing328 Sure. I've tested it manually with the jersey annotations by overwriting the templates. Worked for me.

I've created a pull request. Unfortunatly I've no linux system at the moment so I was not able to fulfill the second bullet on the PR checklist. see #4455

**Comment by wing328** [comment]
Closed via #4455
